### DISCUSSION TITLE
1.7.0.0-rc1

### DISCUSSION THREAD

**Discussion by Jamminroot** [Jamminroot/intag#21] [General]
**FEATURE**: Add Acrylic effect with windows accent
**FEATURE**: Use in-memory data exchange, instead of batch file

![image](https://github.com/Jamminroot/intag/assets/6387654/cbbf37fd-e85b-4cdb-94b8-cbe0fd075dde)

---

**NOTE:** This is a release-candidate (or something like that), as I am not 100% sure about the changes I made. So, for now I am releasing it like this - changes I made aside, that is a regular update.

Please check if you like the changes, and come back to the discussion if there's anything you'd like to revert. 

<hr /><em>This discussion was created from the release <a href='https://github.com/Jamminroot/intag/releases/tag/1.7.0.0-rc1'>1.7.0.0-rc1</a>.</em>

**Comment by montoner0** [comment]
Hi, I've checked the rc and the background looks nice but in my case the tags text is rather low contrast especially with a white-backgrounded window beneath. It would be nice to have some option for the colors or just make them more contrast.
![2023-11-07 23_02_51](https://github.com/Jamminroot/intag/assets/13542804/6dae78a2-e2f5-40cd-87a2-a74cbe2a555c)

**Reply by Jamminroot** [reply]
Thanks for the input. Can you share the color settings of your OS so that I can figure the fix out please?
![image](https://github.com/Jamminroot/intag/assets/6387654/9084dd26-d0f4-455a-a594-2c42824cbc5c)


Also, the formatting seems a bit off with the length of the path. Wouldn't it be nice to have it line wrapped or trimmed? Or maybe cut-in-between like "C:\temp\this...._end.jpg"?

**Reply by montoner0** [reply]
Here it is 
<details>
<summary>Screenshots</summary>

![image](https://github.com/Jamminroot/intag/assets/13542804/518dc758-4041-4e1d-844f-0453c5244fd7)
![image](https://github.com/Jamminroot/intag/assets/13542804/c7b2337f-1cbb-484c-ac5f-272019c274f1)

</details>

> Wouldn't it be nice to have it line wrapped or trimmed? Or maybe cut-in-between like "C:\temp\this...._end.jpg"?

I'm OK with either approach. Though if shortened it would be nice to have a tooltip with a whole path.

**Reply by Jamminroot** [reply]
Thanks. I'll check this out by the end of the week

**Reply by Jamminroot** [reply]
Weirdly, the thing works fine for me:
![image](https://github.com/Jamminroot/intag/assets/6387654/a7ced4c5-158a-4be1-baa8-5f464aa59217)

I'll invest a bit more time into it, and post a test build here for you to see if it works for you, later today

**Reply by montoner0** [reply]
Thanks. Meanwhile, I've noticed another issue: the window is pretty laggy when dragged.

**Reply by Jamminroot** [reply]
That is quite odd, too... Would you mind sharing more input on your build? cpu-gpu in specific. Also, what is your windows build version?
And the .net runtime version used

**Reply by montoner0** [reply]
It's Intel Core i5-3470 + AMD Radeon R9 270. Rather old build but still decent. Windows 10 is 22H2.
I've also tested it on my quite recent laptop (AMD Ryzen 5 5500U + internal GPU) and see the similar issue with window dragging.

I have a bunch of runtimes installed:
```
>dotnet.exe --list-runtimes
Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.0-rc.2.22476.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.29 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.0-rc.2.22472.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.29 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.24 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.0-rc.2.22472.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Reply by montoner0** [reply]
I also remembered that I have this [tweak](https://winaero.com/increase-taskbar-transparency-level-in-windows-10) enabled, it could be related probably.

**Reply by Jamminroot** [reply]
I've repeated the tweak (neat one btw :D ), and checked the dragging - still fine for me.
Use this build (it has a bit of logging enforced), and see the output - it should return the version of the runtime used, disregarding all installed options.
Additionally, I've changed one value in the way transparency is handled, check if that helped with visibility, and "soft" trimmed the object path

[intag.zip](https://github.com/Jamminroot/intag/files/13328928/intag.zip)

**Reply by montoner0** [reply]
Not much better, unfortunately. Now active and inactive tags looks almost the same:
![image](https://github.com/Jamminroot/intag/assets/13542804/31c00e2c-f1ed-4512-b0e1-8bc933f75c33)
Runtime is 6.0.24, no other valuable info in the log.

**Reply by Jamminroot** [reply]
[intag.zip](https://github.com/Jamminroot/intag/files/13358762/intag.zip)
Would you mind checking this set of values?

**Reply by Jamminroot** [reply]
It should increase brightness for "enabled" tags, while leaving disabled at 0.5 (reverted most recent changes). Also I've increased the opacity of the background, so It should let a bit less "light" through

**Reply by montoner0** [reply]
That's better though I'd still prefer to chose another colors. Probably support for a white theme could improve the situation.
Comparison between the 1.7rc and 1.5
![image](https://github.com/Jamminroot/intag/assets/13542804/6c456f6d-e618-4b60-abf0-6a3da3497533)![image](https://github.com/Jamminroot/intag/assets/13542804/39d66523-3f23-4e8f-bb36-565717ac67a1)

**Comment by cachai** [comment]
I remember being so excited that there is a person awesome enough to put this idea into reality(I didn't like the separate 3rd party tagging solutions) but then it was kinda laggy and gave me error messages repeatedly when I'm trying to edit tags of more than 35 folders or something. Now that I came back for this new version, it is so much faster and can edit more than 100 folders at a time with no problem. Thank you so much for the app and you effort to improve it!!

**Reply by Jamminroot** [reply]
Thanks for your kind words!
